projects/plugins/jetpack/extensions/blocks/podcast-player/edit.js

```diff
@@ -58,6 +60,70 @@ const supportUrl =
 		? 'http://en.support.wordpress.com/wordpress-editor/blocks/podcast-player-block/'
 		: 'https://jetpack.com/support/jetpack-blocks/podcast-player-block/';
 
+const actions = {
+	EDIT_URL: 'EDIT_URL',
+	FINISH_EDITING: 'FINISH_EDITING',
+	START_EDITING: 'START_EDITING',
+	SELECT_EPISODE: 'SELECT_EPISODE',
+	FEED_RECEIVED: 'FEED_RECEIVED',
+	CLEAR_FEED: 'CLEAR_FEED',
+	MAKE_INTERACTIVE: 'MAKE_INTERACTIVE',
+	PREVENT_INTERACTIONS: 'PREVENT_INTERACTIONS',
+};
+
+const podcastPlayerReducer = ( state, action ) => {
+	switch ( action.type ) {
+		case actions.EDIT_URL:
+			return {
+				...state,
+				editedUrl: action.payload,
+				selectedGuid: null,
+				feedData: {
+					...state.feedData,
+					options: null,
+				},
+			};
+		case actions.SELECT_EPISODE:
+			return {
+				...state,
+				selectedGuid: action.payload,
+			};
+		case actions.START_EDITING:
+			return {
+				...state,
+				isEditing: true,
+			};
+		case actions.FINISH_EDITING:
+			return {
+				...state,
+				editedUrl: action.payload,
+				isEditing: false,
+			};
+		case actions.FEED_RECEIVED:
+			return {
+				...state,
+				feedData: action.payload,
+			};
+		case actions.CLEAR_FEED:
+			return {
+				...state,
+				feedData: {},
+			};
+		case actions.MAKE_INTERACTIVE:
+			return {
+				...state,
+				isInteractive: true,
+			};
+		case actions.PREVENT_INTERACTIONS:
+			return {
+				...state,
+				isInteractive: false,
+			};
+		default:
+			return { ...state };
+	}
+};
+
```

Comment:
I think this reducer and related code should be in it's own file.

Reply:
Good idea, I'll refactor that.